### PR TITLE
Feature/okta operations

### PR DESCRIPTION
**Pull request by Perdiga**
I changed the folder structure so it is easy to find the operation and work only on that operation
All operations from okta are implemented

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,155 @@
-# flutter_okta_sdk
+# Flutter Okta SDK
```
Comment (Perdiga) [inline README.md:1]: Need to update this readme file

File: android/src/main/kotlin/com/sonikro/flutter_okta_sdk/FlutterOktaSdkPlugin.kt [hunk 2]
```diff
@@ -194,4 +66,84 @@ class FlutterOktaSdkPlugin : FlutterPlugin, MethodCallHandler,
     override fun onDetachedFromActivityForConfigChanges() {
         this.mainActivity = null
     }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        val arguments = call.arguments<Map<String, Any>>()
+        PendingOperation.init(call.method, result)
+
+//        if(!enumContains<Errors>(call.method) ){
```
Comment (Perdiga) [inline android/src/main/kotlin/com/sonikro/flutter_okta_sdk/FlutterOktaSdkPlugin.kt]: Need to put this back, to validate if the method is avaliable

File: android/src/main/kotlin/com/sonikro/flutter_okta_sdk/FlutterOktaSdkPlugin.kt [hunk 3]
```diff
@@ -44,137 +35,18 @@ class FlutterOktaSdkPlugin : FlutterPlugin, MethodCallHandler,
...
-
     override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
         channel.setMethodCallHandler(null)
     }
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
-        webClient!!.handleActivityResult(requestCode, resultCode, data)
-        return pendingOperation != null
+        OktaClient.getWebClient().handleActivityResult(requestCode, resultCode, data)
+        //oktaConfig!!.webClient.handleActivityResult(requestCode, resultCode, data)
```
Comment (Perdiga) [inline android/src/main/kotlin/com/sonikro/flutter_okta_sdk/FlutterOktaSdkPlugin.kt]: Remove this comment

File: android/src/main/kotlin/com/sonikro/flutter_okta_sdk/okta/entities/Constants.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.sonikro.flutter_okta_sdk.okta.entities
```
Comment (Perdiga) [inline android/src/main/kotlin/com/sonikro/flutter_okta_sdk/okta/entities/Constants.kt:1]: I think that we can create objects for the responses, so we will not need this

File: android/src/main/kotlin/com/sonikro/flutter_okta_sdk/okta/operations/IntrospectToken.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        else -> {
+            throw Error(Errors.ERROR_TOKEN_TYPE.errorMessage)
+        }
+    }
+
+    OktaClient.getWebClient().sessionClient.introspectToken(token, tokenName, object : RequestCallback<IntrospectInfo, AuthorizationException> {
+        override fun onSuccess(result: IntrospectInfo) {
+            val params = mutableMapOf<Any, Any?>()
+
+            //TODO: Create a data type for this
```
Comment (Perdiga) [inline android/src/main/kotlin/com/sonikro/flutter_okta_sdk/okta/operations/IntrospectToken.kt:40]: stop using constants and create a proper return object

File: android/src/main/kotlin/com/sonikro/flutter_okta_sdk/okta/operations/RefreshTokens.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import com.sonikro.flutter_okta_sdk.okta.entities.PendingOperation
+
+//import jdk.nashorn.internal.runtime.regexp.joni.constants.Arguments
+
+
+fun refreshTokens() {
+    OktaClient.getWebClient().sessionClient.refreshToken(object : RequestCallback<Tokens, AuthorizationException> {
+        override fun onSuccess(result: Tokens) {
+            val params = mutableMapOf<Any, Any?>()
+            params[Constants.ACCESS_TOKEN_KEY] = result.accessToken
```
Comment (Perdiga) [inline android/src/main/kotlin/com/sonikro/flutter_okta_sdk/okta/operations/RefreshTokens.kt:19]: stop using constants and create a proper return object

**Review by sonikro — APPROVED** [review]
Your nickname should be PerdiGod.

Approved

**Pull request merged**